lib/src/command/uploader.dart

```diff
@@ -28,7 +27,7 @@ class UploaderCommand extends PubCommand {
 
   UploaderCommand() {
     argParser.addOption('server',
-        defaultsTo: cache.sources.hosted.defaultUrl,
+        defaultsTo: cache.sources.hosted.frontFacingDefaultUrl,
         help: 'The package server on which the package is hosted.');
```

Comment:
Since this has never worked before, I propose that we remove this option.

Third-party servers can simply allow users to add/remove uploaders through a web interface.
In fact, I think it's important that we do not make any assumptions about the authorization schemes employed by 3rd-party pub servers.

For corporate internal servers it seems most likely that you'll hook up access to your SSO provider, and/or manage who can publish what packages using a config file.

-----------------------

Long term, I think we are planning to deprecated the `pub uploader` command too. For pub.dev, it no-longer makes sense when the package is owned by a _verified publisher_. This work is tracked in https://github.com/dart-lang/pub-dev/issues/3381

Reply:
I totally agree that the uploader feature is absolutely of no value for third party servers. I only added this after I saw tests where failing because of the changed notation from `pub.dartlang.org` to `pub.dev`. However, when I wanted to change this I found this code where `defaultsTo: cache.sources.hosted.defaultUrl` actually calls

```
  /// Gets the default URL for the package server for hosted dependencies.
  String get defaultUrl {
    // Changing this to pub.dev raises the following concerns:
    //
    //  1. It would blow through users caches.
    //  2. It would cause conflicts for users checking pubspec.lock into git, if using
    //     different versions of the dart-sdk / pub client.
    //  3. It might cause other problems (investigation needed) for pubspec.lock across
    //     different versions of the dart-sdk / pub client.
    //  4. It would expand the API surface we're committed to supporting long-term.
    //
    // Clearly, a bit of investigation is necessary before we update this to
    // pub.dev, it might be attractive to do next time we change the server API.
    return _defaultUrl ??= _pubHostedUrlConfig() ?? 'https://pub.dartlang.org';
  }
```

As I read the notes, I decided it was not a good idea to change this :-)

A quick fix would be that in the mentioned changed line, I just put 'https://pub.dev'. Only when the `PUB_HOSTED_URL` variable is set, this would output wrong info. 
I could change this by using the same evaluation directly as in the getter, but then I have to make the `_pubHostedUrlConfig` public to evaluate. 

Again, I understand it's an option you don't want to support for third party servers; so just setting `http:\\pub.dev` would be an option OR leave this help output to what it is today and revert the unit test where it checks for the correct url. What do you think?